Meta/check-png-sizes.sh

```diff
@@ -0,0 +1,42 @@
+#!/usr/bin/env bash
+
+set -eo pipefail
+
+# How many bytes optipng has to be able to strip out of the file for the optimization to be worth it. The default is 2 KiB.
+MINIMUM_OPTIMIZATION_BYTES="${MINIMUM_OPTIMIZATION_BYTES:-1024}"
+
+script_path=$(cd -P -- "$(dirname -- "$0")" && pwd -P)
+cd "${script_path}/.."
+
+if ! command -v optipng >/dev/null ; then
+    echo 'optipng is not installed, skipping png size check.'
+    echo 'Please install optipng for your system to run this check.'
+    exit 0
+fi
+
+files=()
+for file in "$@"; do
+    if [[ "${file}" == *".png" ]]; then
+        files+=("${file}")
+    fi
+done
+
+if (( ${#files[@]} )); then
+    # We need to allow optipng to write output so we can check what it actually did. We use a dummy file that's discarded afterwards.
+    optimizations=$( printf '%s\0' "${files[@]}" |\
+        xargs -0 -n1 optipng -out dummy-optipng-output.png -clobber 2>&1 |\
+        grep -i -e 'Output IDAT size =' |\
+        sed -E 's/Output IDAT size = [0-9]+ byte(s?) \(([0-9]+) byte(s?) decrease\)/\2/g;s/Output IDAT size = [0-9]+ byte(s?) \(no change\)/0/g' |\
```

Comment:
OptiPNG has not had any updates since 2017 so that seems unlikely, but I understand. It's just that there's no way of getting machine-readable info out of that program.